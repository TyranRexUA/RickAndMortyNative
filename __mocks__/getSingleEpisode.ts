export const getEpisodeWithID1 = {
  "data": {
    "episode": {
      "id": "1",
      "name": "Pilot",
      "air_date": "December 2, 2013",
      "characters": [
        {
          "id": "1",
          "name": "Rick Sanchez",
          "image": "https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        },
        {
          "id": "2",
          "name": "Morty Smith",
          "image": "https://rickandmortyapi.com/api/character/avatar/2.jpeg"
        },
        {
          "id": "35",
          "name": "Bepisian",
          "image": "https://rickandmortyapi.com/api/character/avatar/35.jpeg"
        },
        {
          "id": "38",
          "name": "Beth Smith",
          "image": "https://rickandmortyapi.com/api/character/avatar/38.jpeg"
        },
        {
          "id": "62",
          "name": "Canklanker Thom",
          "image": "https://rickandmortyapi.com/api/character/avatar/62.jpeg"
        },
        {
          "id": "92",
          "name": "Davin",
          "image": "https://rickandmortyapi.com/api/character/avatar/92.jpeg"
        },
        {
          "id": "127",
          "name": "Frank Palicky",
          "image": "https://rickandmortyapi.com/api/character/avatar/127.jpeg"
        },
        {
          "id": "144",
          "name": "Glenn",
          "image": "https://rickandmortyapi.com/api/character/avatar/144.jpeg"
        },
        {
          "id": "158",
          "name": "Hookah Alien",
          "image": "https://rickandmortyapi.com/api/character/avatar/158.jpeg"
        },
        {
          "id": "175",
          "name": "Jerry Smith",
          "image": "https://rickandmortyapi.com/api/character/avatar/175.jpeg"
        },
        {
          "id": "179",
          "name": "Jessica",
          "image": "https://rickandmortyapi.com/api/character/avatar/179.jpeg"
        },
        {
          "id": "181",
          "name": "Jessica's Friend",
          "image": "https://rickandmortyapi.com/api/character/avatar/181.jpeg"
        },
        {
          "id": "239",
          "name": "Mr. Goldenfold",
          "image": "https://rickandmortyapi.com/api/character/avatar/239.jpeg"
        },
        {
          "id": "249",
          "name": "Mrs. Sanchez",
          "image": "https://rickandmortyapi.com/api/character/avatar/249.jpeg"
        },
        {
          "id": "271",
          "name": "Principal Vagina",
          "image": "https://rickandmortyapi.com/api/character/avatar/271.jpeg"
        },
        {
          "id": "338",
          "name": "Summer Smith",
          "image": "https://rickandmortyapi.com/api/character/avatar/338.jpeg"
        },
        {
          "id": "394",
          "name": "Davin",
          "image": "https://rickandmortyapi.com/api/character/avatar/394.jpeg"
        },
        {
          "id": "395",
          "name": "Greebybobe",
          "image": "https://rickandmortyapi.com/api/character/avatar/395.jpeg"
        },
        {
          "id": "435",
          "name": "Pripudlian",
          "image": "https://rickandmortyapi.com/api/character/avatar/435.jpeg"
        }
      ]
    }
  }
}