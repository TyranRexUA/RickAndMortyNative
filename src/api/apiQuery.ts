import gql from "graphql-tag"

export const getEpisodesQuery = gql`
    query getEpisodesQuery($page: Int!, $name: String){
        episodes(page: $page, filter: {name: $name}) {
            info {
                count
                pages
                next
                prev
            }
            results {
                name
                id
            }
        }
    }
`

export const getSingleEpisodeQuery = gql`
    query getEpisodesQuery($id: ID!){
        episode(id: $id) {
            id
            name
            air_date
            characters {
                id
                name
                image
            }
        }
    }
`

export const getCharactersQuery = gql`
    query getEpisodesQuery($page: Int!){
        characters(page: $page) {
            info {
                count
                pages
                next
                prev
            }
            results {
                name
                id
            }
        }
    }
`

export const getSingleCharacterQuery = gql`
    query getEpisodesQuery($id: ID!){
        character(id: $id) {
            id
            name
            image
            species
            gender
            episode {
                name
                id
            }
        }
    }
`