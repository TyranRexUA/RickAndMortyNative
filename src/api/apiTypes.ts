export interface getEpisodesQueryType {
    episodes: {
        info: dataInfoType
        results: episodeType[]
    }
}
export interface dataInfoType {
    "count": number
    "pages": number
    "next": null | number
    "prev": null | number
}


export interface getSingleEpisodeType {
    "episode": {
        "id": string
        "name": string
        "air_date": string
        "characters": singleEpisodeCharacterType[]
    }
}
export interface singleEpisodeCharacterType {
    "id": string
    "name": string
    "image": string
}

export interface getSingleCharacterQueryType {
    "character": {
        "id": string
        "name": string
        "image": string
        "species": string
        "gender": string
        "episode": episodeType[]
    }
}

export interface episodeType {
    "name": string
    "id": string
}