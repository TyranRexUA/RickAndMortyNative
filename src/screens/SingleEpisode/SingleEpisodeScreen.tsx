import React from 'react'
import { useQuery } from "@apollo/client";
import { useRoute } from "@react-navigation/core";
import { getSingleEpisodeQuery } from "../../api/apiQuery";
import Preloader from '../../components/Preloader/Preloader';
import { getSingleEpisodeType } from '../../api/apiTypes';
import { EpisodesStackNavigatorParams } from '../../types/types';
import { RouteProp } from '@react-navigation/native';
import SingleEpisodeView from './SingleEpisodeView';

const SingleEpisodeScreen = () => {
    const route = useRoute<RouteProp<EpisodesStackNavigatorParams, 'SingleEpisode'>>()
    const { loading, error, data } = useQuery<getSingleEpisodeType, { id: string }>(getSingleEpisodeQuery, { variables: { id: route.params.id } });

    if (loading) return (<Preloader />)
    if (data) return (<SingleEpisodeView episode={data?.episode}/>)
    return (<Preloader />)
}

export default SingleEpisodeScreen;