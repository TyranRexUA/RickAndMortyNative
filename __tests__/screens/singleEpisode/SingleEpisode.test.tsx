import React from 'react';
import { render, act, fireEvent, cleanup, RenderAPI } from '@testing-library/react-native';
import { MockedProvider } from '@apollo/client/testing';
import { getSingleEpisodeQuery } from '../../../src/api/apiQuery';
import { getEpisodeWithID1 } from '../../../__mocks__/getSingleEpisode';
import { mockedNavigate } from '../../settings/setupTests';
import SingleEpisodeScreen from '../../../src/screens/SingleEpisode/SingleEpisodeScreen';

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
      ...actualNav,
      useNavigation: () => ({
        navigate: mockedNavigate,
        dispatch: jest.fn(),
      }),
    };
  });

describe('EpisodesItem', () => {
    let wrapper: RenderAPI;
    const promise = Promise.resolve()

    const episodeMock = {
        request: {
            query: getSingleEpisodeQuery,
            variables: { id: "1" },
        },
        result: getEpisodeWithID1,
    };

    const errorMock = {
        request: {
            query: getSingleEpisodeQuery,
            variables: { id: "1" },
        },
        error: new Error(),
    }

    afterEach(cleanup)

    beforeEach(async () => {
        wrapper = render(
            <MockedProvider mocks={[episodeMock]}>
                <SingleEpisodeScreen />
            </MockedProvider>
        );
        await act(async () => promise)
    });

    test('with loader', async () => {
        await act(async () => {
            await expect(wrapper.getByTestId('Loader')).toBeTruthy();
        })
    })

    test('With error', async () => {
        const errorWrapper = render(
            <MockedProvider mocks={[errorMock]}>
                <SingleEpisodeScreen />
            </MockedProvider>
        );
        await act(async () => promise)
        expect(errorWrapper.getByText(/error/i)).toBeTruthy();
    })

    describe('with data', () => {

        beforeEach(async () => {
            await act(() => promise)
        });

        test('check Episodes', async () => {
            expect(wrapper.getAllByTestId('CharactersItem').length).toBe(19);
        });

        test('check navigation', async () => {
            await act(async () => {
                await fireEvent(wrapper.getAllByTestId('CharactersItem')[0], 'onPress')
            })
            expect(mockedNavigate).toHaveBeenCalled();
        })

        test('Snapshot checking', async () => {
            expect(wrapper.toJSON()).toMatchSnapshot();
        })
    })

});