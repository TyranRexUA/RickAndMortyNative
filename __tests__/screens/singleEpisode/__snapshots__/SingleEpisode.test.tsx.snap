// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodesItem with data Snapshot checking 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 10,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "flex": 1,
          "fontSize": 20,
          "fontWeight": "bold",
          "marginTop": 20,
          "textAlign": "center",
        }
      }
    >
      Pilot
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "marginTop": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "marginRight": 10,
          }
        }
      >
        AIR DATE
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        December 2, 2013
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "marginTop": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "marginRight": 10,
          }
        }
      >
        CHARACTERS
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Rick Sanchez
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/2.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Morty Smith
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/35.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Bepisian
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/38.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Beth Smith
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/62.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Canklanker Thom
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/92.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Davin
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/127.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Frank Palicky
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/144.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Glenn
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/158.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Hookah Alien
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/175.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Jerry Smith
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/179.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Jessica
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/181.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Jessica's Friend
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/239.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Mr. Goldenfold
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/249.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Mrs. Sanchez
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/271.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Principal Vagina
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/338.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Summer Smith
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/394.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Davin
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/395.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Greebybobe
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="CharactersItem"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingVertical": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://rickandmortyapi.com/api/character/avatar/435.jpeg",
              }
            }
            style={
              Object {
                "borderRadius": 100,
                "height": 30,
                "marginRight": 10,
                "width": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "flex": 1,
                "fontSize": 16,
              }
            }
          >
            Pripudlian
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "rgba(0, 0, 0, 0.12)",
                "height": 0.5,
              },
              undefined,
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
